src/Common/MemoryTracker.h

```diff
@@ -103,7 +103,7 @@ class MemoryTracker
     [[nodiscard]] AllocationTrace allocImpl(Int64 size, bool throw_if_memory_exceeded, MemoryTracker * query_tracker = nullptr, double _sample_probability = -1.0);
     [[nodiscard]] AllocationTrace free(Int64 size, double _sample_probability = -1.0);
 public:
-
+    static constexpr Int64 threshold = 16 * 1024 * 1024;   /// The choice is arbitrary (maybe we should decrease it)
```

Comment:
Why mix 2 completely different variables? The original threshold was to throw logical errors in debug mode, but now you are using it to update memory credits too, so we can't change one without changing the other.

Reply:
Changes were made! I misunderstand the relationship between the two variables.